### ISSUE TITLE
[Suggestion] Builder Block Placement Tool

### ISSUE THREAD

**Issue by McGaiser** [McJtyMods/RFTools#1762]
Create a tool to improve the interface to the builder block.

I use the builder block and its shape card to sculpt terrain, but the interface is really painful. It usually consists of setting your shape card with the desired shape and size, then making an initial guess as to where it will be placed in relation to the block. You put the card into the block, preview the area, figure out its not right, take the shape card out of the block, adjust location offset, put card back into the block, preview... rinse/repeat. 

What I suggest would be to have a tool/wand/gun that you could link to the builder block. It can be aimed at a location and "fired". The location that it hits would serve as the center of the building block. It would also automatically active the preview so you could quickly see where the shape would be. There could then be some option to active the block and build the shape, or you can fire the gun again and reposition the shape. The would allow people to fly around their terrain/sculpture and reshape it as desired similar to how the VoxelSniper mod works but in a survival context.

Giving the ability to change the location, shape and size without having to always return to the builder block would be simply amazing.

It would also be nice if the builder block could be configured to pull from a certain side when creating shapes, and expel removed blocks to another side when excavating.

This tool would be a game changer to me as I love how you have implemented dev style tools while keeping things balanced in modded survival play.

**Comment by josephcsible** [comment]
If you shift-right-click the Builder with the shape card, you can then right-click two opposite corners of the area, and it will automatically fill in the desired size and offset. Does this cover your use-case?

**Comment by McGaiser** [comment]
No, at best that works for a simple area, but Im mostly using spheres and hemispheres. Have a look at the voxelsniper mod. I want that but within a creative context and the builder block has most of it done already. Its just really really painful to use on large projects.
